R/pkg/inst/worker/daemon.R

```diff
@@ -31,7 +31,15 @@ inputCon <- socketConnection(
     port = port, open = "rb", blocking = TRUE, timeout = connectionTimeout)
 
 while (TRUE) {
-  ready <- socketSelect(list(inputCon))
+  ready <- socketSelect(list(inputCon), timeout = 1)
```

Comment:
BTW, the old behaviour was simply to hang and wait there indefinitely.

https://stat.ethz.ch/R-manual/R-devel/library/base/html/socketSelect.html

> numeric or NULL. Time in seconds to wait for a socket to become available; NULL means wait indefinitely.